lib/drivers/weitai/weiTaiUSB.js

```diff
@@ -0,0 +1,1625 @@
+          if (senList[c] !== senList[senLgtn - 1]) {
+            insulinSensitivity.push({
+              amount: senList[c] / 10,
+              start: c * 30 * 60 * 1000,
+            });
+          }
+        }
+      }
+
+      // insulinSensitivity
+      const bgList = settings.lowBg;
+      const bgHighList = settings.highBg;
+      if (bgHighList.length > 1) {
+        // remove default value
+        bgList.pop();
+        bgHighList.pop();
+      }
+      const bgTarget = [];
+      for (let c = 0; c < bgList.length; c++) {
+        if (c === 0) {
+          bgTarget.push({
+            start: 0,
+            low: parseInt(bgList[0] / 10, 10),
+            high: parseInt(bgHighList[0] / 10, 10),
+          });
+        } else {
+          const targetLength = bgTarget.length;
+          const mergerLow = bgList[c] / 10;
+          const mergerHigh = bgHighList[c] / 10;
+          if (mergerLow !== bgTarget[targetLength - 1].low || mergerHigh !== bgTarget[targetLength - 1].high) {
+            bgTarget.push({
+              start: 30 * 60 * 1000 * c,
+              low: mergerLow,
+              high: mergerHigh,
+            });
+          }
+        }
+      }
+
+      // basalSchedules
+      const basalSchedules = {};
+      let activeName = '';
+      for (let i = 0; i < 7; i++) {
+        const nameIndex = 'program' + i + '_name';
+        const name = settings[nameIndex] || 'basal_' + i;
+        if (settings[nameIndex]) {
+          activeName = settings[nameIndex];
+        }
+        const basalList = settings['program' + i] || [];
+        basalSchedules[name] = [];
+        if (basalList.length) {
+          for (let b = 0; b < basalList.length; b++) {
+            if (b === 0) {
+              basalSchedules[name].push({
+                start: 0,
+                rate: basalList[b] * 0.0125,
+              });
+            } else if (basalList[b] !== basalList[b - 1]) {
+              basalSchedules[name].push({
+                start: b * 30 * 60 * 1000,
+                rate: basalList[b] * 0.0125,
+              });
+            }
+          }
+        }
+      }
+      return {
+        carbRatio,
+        insulinSensitivity,
+        bgTarget,
+        basalSchedules,
+        activeName,
+      };
+    },
+
+    buildSettings(settings) {
+      // basalSchedules
+      const settingRes = this.buildBasalSchedules(settings);
+      const postsettings = cfg.builder
+        .makePumpSettings()
+        .with_activeSchedule(settingRes.activeName || 'basal_1')
+        .with_units({ carb: 'grams', bg: 'mg/dL' })
+        .with_basalSchedules(settingRes.basalSchedules)
+        .with_carbRatio(settingRes.carbRatio)
+        .with_insulinSensitivity(settingRes.insulinSensitivity)
+        .with_bgTarget(settingRes.bgTarget)
+        .with_manufacturers(['Microtech'])
+        .with_serialNumber(cfg.deviceInfo.serialNumber)
+        .with_deviceTime(sundial.formatDeviceTime(new Date('2020-02-28 10:00:00').valueOf()))
```

Comment:
@pauloved This should have the current device time at the time of upload, not a random hard-coded time value.